### PR TITLE
[Facebook Gem] Add ability to handle "standby" webhook, add more logging

### PR DESCRIPTION
**Pull request by emaglio**
The best way is to review this is commit by commit:
- ~just a quick fix of some differences in the different module (it can be removed if not required)~
- implementation of the handover protocol: a new `Incoming::Standby` class has been implemented which will return in its methods `events` an array of events in base on incoming data
- ~implementation of the logger, not sure if that is not enough or too much~ 
- updated changelog and readme (the version bump and release it could be done with a single commit directly in master maybe?)

### CODE REVIEW

File: lib/facebook/messenger/incoming/standby.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        #
+        def time
+          Time.at(messaging['time'] / 1000)
+        end
+
+        private
+
+        def build_events
+          messaging['standby'].map do |event|
+            key = (event.keys - %w[sender recipient timestamp]).first
```
Comment (emaglio) [inline lib/facebook/messenger/incoming/standby.rb:52]: this should give us the missing key which will distinguish the different events - is there a better way to do so?

File: README.md [hunk 2]
```diff
@@ -337,11 +337,27 @@ Facebook::Messenger::Profile.set({
...
+
+#### Handle the Handover Protocol
+
+See Facebook's documentation on [Handover Protocol](https://developers.facebook.com/docs/messenger-platform/handover-protocol)
+
+```ruby
+Bot.on :standby do |standby|
+  standby.id                     # => { 'id' => '2015573629214912' }
+  standby.time                   # => 2016-04-22 21:30:36 +0200
+  standby.events.each do |event| # => array of events received in the standby channel which could be
```
Comment (cfurrow) [inline README.md]: I think more code showing what each `event` looks like would be helpful. e.g. what are the attributes for `message_reads` or `messages`, etc. So if someone has to implement this on their server, they could get a good idea of what properties and attributes are available and required for them to use.

File: lib/facebook/messenger/bot.rb [hunk 3]
```diff
@@ -29,147 +30,147 @@ module Bot
...
         payment
         policy_enforcement
+        standby
       ].freeze
 
       BROADCAST_MESSAGE_REGULAR = 'REGULAR'.freeze
       BROADCAST_MESSAGE_SILENT  = 'SILENT_PUSH'.freeze
       BROADCAST_MESSAGE_NO_PUSH = 'NO_PUSH'.freeze
 
-      class << self
```
Comment (cfurrow) [inline lib/facebook/messenger/bot.rb]: While I'm normally in favor of these refactors, I'd advise against large ones like this because as we diverge more and more from the source repo, it makes it harder and harder to pull in their changes into this repo because the diffs are HUGE and have completely unrelated code changes in them.

This entire PR should be smaller, and more focused, with less refactors.

**Review by cfurrow — CHANGES_REQUESTED** [review]
Please limit the changes you are making to JUST the standby protocol additions, and remove the refactors and loggers, etc.

Adding more features/refactors make it very difficult to pull in changes from the upstream source repo (https://github.com/jgorset/facebook-messenger) when there are lots of unrelated/non-functional changes in our repo.

Please revise.

File: lib/facebook/messenger/incoming/standby.rb (hunk 1 above)
Comment (cfurrow) [reply lib/facebook/messenger/incoming/standby.rb:52]: Hm, I think that's ok for now.

But, what if FB adds more keys, and the one we need is not `.first` ? We'd want to always be sure we could handle it appropriately.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 require 'coveralls'
 require 'webmock/rspec'
+require 'pry'
```
Comment (cfurrow) [inline spec/spec_helper.rb]: Still needed?

**Review by cfurrow — APPROVED** [review]
